### DISCUSSION TITLE
FormatJS SWC plugin support

### DISCUSSION THREAD

**Discussion by patroza** [vercel/next.js#33698] [Help]
Hi,

Is there any way to integrate https://formatjs.io/docs/tooling/swc-plugin/ 

Thanks

**Comment by joggienl** [comment]
I managed to work around it by adding the formatjs eslint plugin. It can autofix the ID's and if added, there is no need for te plugin (at this point).

source: https://github.com/vercel/next.js/discussions/30174#discussioncomment-1922431

**Reply by nkpz** [reply]
Thanks, it wasn't immediately clear to me that this was possible based on the docs.

It also wasn't immediately clear how it works, so I hope this helps as a quick resource to someone in the same boat:
Once installed, to get autofix working you'll need to specify in your eslint config the interpolation pattern that would have otherwise been in your .babelrc like so:
```
    "formatjs/enforce-id": [
      "error",
      {
        "idInterpolationPattern": "[sha512:contenthash:base64:6]"
      }
    ],
```

**Reply by khasty720** [reply]
Still a solid work around in 2024

**Comment by webda2l** [comment]
For note, there is now a WASM version of this plugin merged in the formatjs codebase. See https://github.com/formatjs/formatjs/tree/main/packages/swc-plugin-experimental (!= previous https://github.com/formatjs/formatjs/tree/main/packages/swc-plugin)

But it doesn't work as expected.. :/ Some dev resources of the NextJs team will be welcome I think.
See https://github.com/formatjs/formatjs/issues/3589
